### DISCUSSION TITLE
Esp8266 based board has not enough RAM to load the module

### DISCUSSION THREAD

**Discussion by Cinmay2014** [n-elia/MAX30102-MicroPython-driver#18] [Help]
When I tried from max30102 import MAX30102, always got importError: no module named 'max30102.MAX30102'
dir as below:
lib
 -max30102
    --circular_buffer.py
    --__init__.py
I also tried from max30102.__init__ import MAX30102, got another error:
Traceback(most recent call last):
File"<stdin>", line1, in <module>
MemoryError: memory allocation failed,allocating 416 bytes

Please help me.

**Comment by n-elia** [comment]
Hi,
which board are you using? And which tools are you using to copy the files on it?
You can also attach some screenshots if you want.
Also, post the code you're trying to execute, please!

**Comment by Cinmay2014** [comment]
Thanks for your quick reply.
I am using esp8266
And installed the driver as Thonny IDE plug-in, I also tried to install it directly to the board, and got the same issue.

>>> from max30102 import MAX30102
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'max30102'
>>>

**Comment by n-elia** [comment]
Hi,
did you install the latest (v0.4.1) version?
Are you absolutely sure that the files have been correctly written onto non-volatile memory? It seems like it cannot find the file.

May you provide a wifi internet connection to your board? If so, can you try the example suggested in the readme [here](https://github.com/n-elia/MAX30102-MicroPython-driver#1a---network-enabled-micropython-ports) ?

**Comment by Cinmay2014** [comment]
Yes, I installed the latest version.
I tried the manual way and got a memory error. 
I will try the wifi-enable method and let you know the result.

**Comment by n-elia** [comment]
Ok, try that way. 

Remember to reset the board (Ctrl-D in Thonny) and let it run boot.py and main.py, do not play your scripts directly from Thonny.

The online installation works fine, so if you will get the memory error, then it means that you have not enough RAM on your board to run this driver, and you could try to precompile or frozen the driver or try to write a lighter one on your own.

Let me know what happens!

**Comment by Cinmay2014** [comment]
Tried it and got the error:


Trying to connect to the Internet to download the module.
network config: ('192.168.1.216', '255.255.255.0', '192.168.1.1', '192.168.1.98')
Installing to: /lib/
Error installing 'micropython-max30102': memory allocation failed, allocating 15685 bytes, packages may be partially installed


Can you tell me how to frozen the driver?

**Reply by scascor** [reply]
i tried the“upip install”，still failed 。
maybe you can try “mpy-cross”。https://github.com/micropython/micropython/tree/master/mpy-cross。 put the "*.mpy" file  into the esp8266, then you can use it. try！

**Reply by n-elia** [reply]
> i tried the“upip install”，still failed 。 maybe you can try “mpy-cross”。[https://github.com/micropython/micropython/tree/master/mpy-cross。](https://github.com/micropython/micropython/tree/master/mpy-cross%E3%80%82) put the "*.mpy" file into the esp8266, then you can use it. try！

Thanks for your detailed answer!
The link is somehow broken. Here is a working one: [mpy-cross](https://github.com/micropython/micropython/tree/master/mpy-cross) .

**Comment by n-elia** [comment]
Hi,
Then your board has not enough RAM to load the library.
Take a look [here](https://forum.micropython.org/viewtopic.php?t=4510) for instructions on freezing modules.

Edit: I converted this into a discussion, since it's not a library issue. Let me know if you will be able to successfully freeze the module!

**Comment by scascor** [comment]
get a same error.

**Reply by n-elia** [reply]
Hi, unfortunately it appears that your board does not have enough RAM memory to properly load this module. You can either try to freeze the library, as suggested in the previous messages, or to change board (e.g. you could use an ESP8266 with more RAM memory or an ESP32).

**Reply by scascor** [reply]
i have fixed the issue  with “freeze the library”。 thanks。

**Reply by n-elia** [reply]
> i have fixed the issue with “freeze the library”。 thanks。

I am very happy that it worked! Thank you for sharing the solution with the community!

**Comment by n-elia** [comment]
Hey guys,

[here](https://github.com/n-elia/MAX30102-MicroPython-driver/pull/19) you can find a thorough explanation about this issue and how to overcome it.

Cheers!
